### PR TITLE
fix(generics): support generic response types

### PR DESCRIPTION
**Pull request by mharkins-cosm**
This is an attempt to address issue #204.  Because the nested generic objects must be checked dynamically for names, I had to get rid of the `NAME` constant and replace it with a `name()` method.  Then it's just including named type parameter names in the parent name surrounded by angle brackets (`<>`).

### CODE REVIEW

**Comment by tiagolobocastro** [comment]
@mharkins-cosm could you please rebase so we can run the tests with gha?
I think this would be a breaking change to the API, so we'd have to bump up the version and add this to the changelog.

**Comment by mharkins-cosm** [comment]
> > > @mharkins-cosm could you please rebase so we can run the tests with gha? I think this would be a breaking change to the API, so we'd have to bump up the version and add this to the changelog.
> > 
> > 
> > Done
> 
> Cool, it's running now. Could you please also add tests for generic response types and also update the changelog? Thanks

Yes, I will.  I'm sorry I haven't had time to get to it yet.  It's on my todo list!

**Comment by mharkins-cosm** [comment]
@tiagolobocastro Finally got to adding the requested test and changelog.  Would you mind having a look and kicking of the checks if it looks good to you?

**Comment by mharkins-cosm** [comment]
saw the lint check failed.  i think i should have that fixed now.

File: macros/src/actix.rs [hunk 1]
```diff
@@ -748,7 +748,20 @@ pub fn emit_v2_definition(input: TokenStream) -> TokenStream {
...
+    let base_name = extract_rename(&item_ast.attrs).unwrap_or_else(|| name.to_string());
+    let type_params: Vec<&Ident> = generics.type_params().map(|p| &p.ident).collect();
+    let schema_name = if type_params.is_empty() {
+        quote! { #base_name }
+    } else {
+        let type_names = quote! {
+            vec![#(#type_params::name()),*]
+                .iter()
+                .filter(|n| n.is_some())
+                .map(|n| n.as_ref().unwrap().clone()).collect::<Vec<String>>()
```
Comment (tiagolobocastro) [inline macros/src/actix.rs:760]: could use filter_map here? also maybe into_iter instead of iter avoiding having to write the clone?

**Comment by tiagolobocastro** [comment]
> saw the lint check failed. i think i should have that fixed now.

cool I kicked the GHA, will have a another look tomorrow

File: macros/src/actix.rs (hunk 1 above)
Comment (mharkins-cosm) [reply macros/src/actix.rs:760]: hmmm... i get a warning for `into_iter`, but can use `filter_map`.

Comment (mharkins-cosm) [reply macros/src/actix.rs]: ✅

**Comment by mharkins-cosm** [comment]
Funny the clippy warning is not part of this PR, nor does it appear to be accurate.  Do you want me to add an `allow` annotation anyway, to allow the check to pass?

**Comment by tiagolobocastro** [comment]
> Funny the clippy warning is not part of this PR, nor does it appear to be accurate. Do you want me to add an `allow` annotation anyway, to allow the check to pass?

That'd be great if you don't mind 👍

**Comment by mharkins-cosm** [comment]
Thanks very much @tiagolobocastro ! Any idea when it'll be released, along with the actix 4 support?

**Comment by tiagolobocastro** [comment]
No, thank you! I'll probably try to publish it this weekend.

**Pull request merged**